controllers/login.js

```diff
@@ -286,14 +286,29 @@ router.get('/login/success', authHelper.authChecker, async (req, res, next) => {
 	return null;
 });
 
+const sessionDestroyer = (req, res, rej, next) => {
+	if (req.url === '/logout') {
+		req.session.destroy((err) => {
+			if (err) {
+				rej(`Error destroying session: ${err}`);
+			} else {
+				// clear the CSRF token to prevent re-use after logout
+				res.locals.csrfToken = null;
+			}
+		});
+	}
+	return next();
+};
+
 router.get('/logout/', (req, res, next) => {
 	api(req)
 		.del('/authentication') // async, ignore result
 		.catch((err) => {
 			logger.error('error during logout.', formatError(err));
 		});
-	return authHelper.clearCookie(req, res, { destroySession: true })
-		.then(() => res.redirect('/'))
+
+	return authHelper.clearCookie(req, res, sessionDestroyer)
+		.then(() => res.redirect(`/?rand=${Math.random()}`))
```

Comment:
Why we need it?
?rand=${Math.random()}

Reply:
Yes, by adding this random number to the URL as a query parameter, it ensures that each time the page is redirected, the URL will be unique, this preventing any caching issues in the future.